### PR TITLE
feat: Multi cluster routing

### PR DESCRIPTION
**Pull request by danieljbruce**
This PR allows users of the client library to provide a list of clusters in an API request so that routing of requests will stay restricted to that group of clusters.

### CODE REVIEW

File: system-test/common.ts
Comment (kolea2) [inline system-test/common.ts:21]: same question here

File: src/app-profile.ts
Comment (kolea2) [inline src/app-profile.ts:32]: The Java client uses a set instead of an array. Is there a compelling reason to have an array here? If no, can we switch to a set?

**Review by kolea2 — CHANGES_REQUESTED** [review]

File: src/app-profile.ts
Comment (danieljbruce) [reply src/app-profile.ts:32]: Sounds like a good plan. I can change to a set. The only reason we would want to use an array is if we want for some reason for the order to be preserved every time we make a call to get the clusters in an app profile. I can't see why we would care about that though so a set is fine.

File: system-test/common.ts
Comment (danieljbruce) [reply system-test/common.ts:19]: These variables were moved into a file called common.js because now two files use them. `app-profile.ts` is one of them and the other is `instance.ts`.

File: system-test/common.ts
Comment (danieljbruce) [reply system-test/common.ts:21]: These variables were moved into a file called common.js because now two files use them. `app-profile.ts` is one of them and the other is `instance.ts`.

File: system-test/app-profile.ts
Comment (danieljbruce) [reply system-test/app-profile.ts:64]: Sounds good. I'll go make this change.

File: src/app-profile.ts
Comment (danieljbruce) [reply src/app-profile.ts:32]: The only consideration is that in the response object we get an array so maybe we want an array input for consistency, but probably it doesn't matter.

File: system-test/app-profile.ts
Comment (danieljbruce) [reply system-test/app-profile.ts:21]: Added these tests. In the process of adding these tests it looks like a lot of restructuring was required in order to eliminate duplicate code fragments so let me know if you have any questions regarding the changes.

File: src/app-profile.ts
Comment (kolea2) [inline src/app-profile.ts:32]: a thought - it might be nice to provide an additional method which takes in a set of strings - we don't need the cluster object anywhere, so this could be easier on the user rather than having to construct a Cluster object.

File: test/instance.ts
Comment (kolea2) [inline test/instance.ts:260]: this this needs a description and change to "set"

File: system-test/app-profile.ts
Comment (danieljbruce) [reply system-test/app-profile.ts:113]: Good point. Should I now eliminate the test that creates a single cluster since that test is now exactly the same as the first part of this one?

File: test/instance.ts
Comment (danieljbruce) [reply test/instance.ts]: Changed to say `set`. Regarding description, the other tests in the same describe block say `'a cluster value'` and `"an 'any' value"`.  Should those descriptions change along with this to something like `should accept an 'any' value`?

File: src/app-profile.ts
Comment (kolea2) [inline src/app-profile.ts:238]:
```suggestion
          'An app profile routing policy can only contain "any" for multi cluster routing, a `Cluster` for single routing,  or a set of clusterIds as strings or `Clusters` for mutli cluster routing.'
```

File: system-test/app-profile.ts
Comment (kolea2) [reply system-test/app-profile.ts:124]: I would keep existing tests as is for now. We can do cleanup later

File: src/app-profile.ts
Comment (kolea2) [inline src/app-profile.ts:32]: let's add a test for this new option `Set<string>`

**Review by kolea2 — CHANGES_REQUESTED** [review]

File: protos/protos.d.ts
Comment (danieljbruce) [reply protos/protos.d.ts]: Good point. I think this was generated by `npm run fix` to fix linting issues. Should I go through and manually revert all of these # to | changes?

File: src/app-profile.ts
Comment (danieljbruce) [reply src/app-profile.ts:32]: You can find this under the test that says `should create a profile with multiple clusters using strings`.

**Comment by bcoe** [comment]
Just an FYI, if you're working on a branch from a fork, add `owlbot:run` and `kokoro:force-run` to kick off tests/post-processing.

File: test/instance.ts
Comment (kolea2) [inline test/instance.ts:260]: nit - update this description

**Review by kolea2 — COMMENTED** [review]
lgtm after last small comment

File: test/instance.ts
Comment (danieljbruce) [reply test/instance.ts]: I updated this to say `a set of cluster objects`, but I think you might have wanted something else. In this describe block our description string says, `"should respect the routing option with"` and then each `it` describes what we pass into the routing option. In this case, on the line that says, `const options = {routing: new Set(clusters)};` we are setting the routing option to a set of clusters.

**Review by kolea2 — APPROVED** [review]
lgtm, would appreciate if @bcoe could also sign off before merging :) thanks!

**Review by bcoe — APPROVED** [review]
Left a couple non-blocking recommendations.

File: system-test/common.ts
Comment (danieljbruce) [reply system-test/common.ts:22]: I'll create a separate issue for this since I tried it and there was more cleanup for the sake of getting this PR in and keeping things separate.

Comment (danieljbruce) [reply system-test/common.ts:22]: https://github.com/googleapis/nodejs-bigtable/issues/1084

**Pull request merged**
